packages/core/src/StateMachine.ts

```diff
@@ -629,4 +637,57 @@ export class StateMachine<
 
     return restoredSnapshot;
   }
+
+  /**
+   * Runs an executable action. Executable actions are returned from the
+   * `transition(…)` function.
+   *
+   * @example
+   *
+   * ```ts
+   * const [state, actions] = transition(someMachine, someState, someEvent);
+   *
+   * for (const action of actions) {
+   *   // Executes the action
+   *   someMachine.executeAction(action);
+   * }
+   * ```
+   */
+  public executeAction(
+    action: ExecutableActionObject,
+    actor: AnyActorRef = createEmptyActor()
+  ) {
+    const actorScope = (actor as any)._actorScope as AnyActorScope;
+    const defer = actorScope.defer;
+    actorScope.defer = (fn) => fn();
+    try {
+      switch (action.type) {
+        case 'xstate.cancel':
+          executeCancel(actorScope, action.params as any);
+          return;
+        case 'xstate.raise':
+          if (typeof (action as any).params.delay !== 'number') {
+            return;
+          }
+          executeRaise(actorScope, action.params as any);
+          return;
+        case 'xstate.sendTo':
+          executeSendTo(actorScope, action.params as any);
+          return;
+      }
```

Comment:
I believe this switch/case misses some of the other actions, like `emit`, `log`. It's also still not completely obvious to me how the user should deal with `spawnChild` and `stopChild` - it would be great if we'd have some tests showing how to interact with those.